DatadogCore/Tests/Datadog/Core/Persistence/FilesOrchestratorTests.swift

```diff
@@ -55,17 +55,20 @@ class FilesOrchestratorTests: XCTestCase {
     }
 
     func testWhenSameWritableFileWasUsedMaxNumberOfTimes_itCreatesNewFile() throws {
-        let orchestrator = configureOrchestrator(using: RelativeDateProvider(advancingBySeconds: 0.001))
+        let dateProvider = DateProviderMock()
+        let orchestrator = configureOrchestrator(using: dateProvider)
```

Comment:
💡 This test started being flaky after adding more calls to `dateProvider.now` in files orchestrator. It is because we were using relative date provider mock that advances the date after each call to `.now`. In general, we should avoid using `RelativeDateProvider` and be more specific about the "now" value required in each test.